wp-content/plugins/tapestry/endpoints.php

```diff
@@ -3,3 +3,73 @@
  * Tapestry Endpoints
  *
  */
+
+require __DIR__ . '/controller/class.user-controller.php';
+
+ // User endpoints
+
+// Get user's progress
+add_action( 'rest_api_init', function () {
+    register_rest_route( 'tapestry-tool/v1', 'users/progress/', array(
+      'methods' => 'GET',
+      'callback' => 'get_user_progress_by_post_id',
+    ) );
+});
+
+// Get user progress on a tapestry page by user id and post id. Will need to pass these as query parameters
+// Example: http://localhost:8888/tapestry-wp/wp-json/tapestry-tool/v1/users/progress?userid=1&postid=42
+function get_user_progress_by_post_id($data) {
+    $userId = $data['userid'];
```

Comment:
Please always use underscores to separate the words so it's more readable. This should be `$data['user_id']` and the one below should be `$data['post_id']`. Please do this for your whole PR and keep this convention for the frontend work as well.

Reply:
changed to user_id on frontend and backend